kolibri/plugins/learn/assets/src/state/actions/recommended.js

```diff
@@ -0,0 +1,272 @@
+} from 'kolibri.coreVue.vuex.getters';
+
+import {
+  samePageCheckGenerator,
+  setChannelInfo,
+  handleApiError,
+} from 'kolibri.coreVue.vuex.actions';
+
+import { PageNames } from '../../constants';
+import { contentState } from './main';
+
+import ConditionalPromise from 'kolibri.lib.conditionalPromise';
+import uniqBy from 'lodash/uniqBy';
+
+function _getPopular(channelId) {
+  const channelPayload = { channel_id: channelId };
+  const popularPayload = { popular: 'true' };
+  return ContentNodeResource.getCollection(channelPayload, popularPayload).fetch();
+}
+
+function _getNextSteps(channelId, state) {
+  const nextStepsPayload = { next_steps: currentUserId(state) };
+  const channelPayload = { channel_id: channelId };
+
+  if (isFacilityUser(state)) {
+    return ContentNodeResource.getCollection(channelPayload, nextStepsPayload).fetch();
+  }
+  return Promise.resolve([]);
+}
+
+function _getResume(channelId, state) {
+  const resumePayload = { resume: currentUserId(state) };
+  const channelPayload = { channel_id: channelId };
+
+  // all you really need is user_id. Should we check in parent?
+  if (isFacilityUser(state)) {
+    return ContentNodeResource.getCollection(channelPayload, resumePayload).fetch();
+  }
+  return Promise.resolve([]);
+}
+
+function _getOverview(channelId, cursor) {
+  const channelPayload = { channel_id: channelId };
+
+  return ContentNodeResource.getAllContentCollection(channelPayload, { cursor }).fetch();
+}
+
+function showLearnChannel(store, channelId, cursor) {
+  // Special case for when only the page number changes:
+  // Don't set the 'page loading' boolean, to prevent flash and loss of keyboard focus.
+  const state = store.state;
+  if (state.pageName !== PageNames.LEARN_CHANNEL || getCurrentChannelId(state) !== channelId) {
+    store.dispatch('CORE_SET_PAGE_LOADING', true);
+  }
+
+  const channelsPromise = setChannelInfo(store, channelId);
+  channelsPromise.only(
+    samePageCheckGenerator(store),
+    () => {
+      if (!getCurrentChannelObject(state)) {
+        router.replace({ name: PageNames.CONTENT_UNAVAILABLE });
+        return;
+      }
+      ConditionalPromise.all([
+        _getNextSteps(channelId, state),
+        _getPopular(channelId),
+        _getResume(channelId, state),
+        _getOverview(channelId, cursor),
+      ]).only(
+        samePageCheckGenerator(store),
+        ([nextSteps, popular, resume, overview]) => {
+          const currentChannel = getCurrentChannelObject(store.state);
+          const pageState = {
+            // Hard to guarantee this uniqueness on the database side, so
+            // do a uniqBy content_id here, to prevent confusing repeated
+            // content items.
+            nextSteps: uniqBy(nextSteps, 'content_id').map(contentState),
+            popular: uniqBy(popular, 'content_id').map(contentState),
+            resume: uniqBy(resume, 'content_id').map(contentState),
+            overview: overview.map(contentState),
```

Comment:
Should probably do a `uniqBy` here, just in case our overview returns repeated items also.